drivers/storage/portworx/portworx.go

```diff
@@ -1214,27 +1215,75 @@ func setAutopilotDefaults(
 			},
 		}
 	}
+}
 
-	// px-3.0.0 -- add PX_ENABLE_TLS=true env to AutoPilot when Security + TLS are enabled
+func setTLSDefaults(
+	toUpdate *corev1.StorageCluster,
+) {
 	sec := toUpdate.Spec.Security
-	if sec != nil && sec.Enabled && sec.TLS != nil && sec.TLS.Enabled != nil && *sec.TLS.Enabled &&
-		pxutil.GetPortworxVersion(toUpdate).GreaterThanOrEqual(pxutil.MinimumPxVersionAutoTLS) {
-		newEnv := make([]v1.EnvVar, 0, len(toUpdate.Spec.Autopilot.Env)+1)
+	secEnabled := true
+
+	if pxutil.GetPortworxVersion(toUpdate).LessThan(pxutil.MinimumPxVersionAutoTLS) {
+		// px version too low for auto-ssl setup, bail out..
+		return
+	} else if sec == nil || !sec.Enabled || sec.TLS == nil || sec.TLS.Enabled == nil || !*sec.TLS.Enabled {
+		// security/TLS not enabled
+		secEnabled = false
+		// proceed to check env-vars
+	}
+
+	type listOpType struct {
+		list    *[]v1.EnvVar
+		enabled bool
+	}
+	var listOps []listOpType
+
+	if toUpdate.Spec.Autopilot != nil {
```

Comment:
it's easier to understand with a sub function and call that function for both autopilot and stork, instead of moving their pointers to the list